pkg/storage/gc/gc.go

```diff
@@ -120,11 +125,18 @@ func (gc GarbageCollect) cleanRepo(repo string) error {
 		return err
 	}
 
-	// update repos's index.json in storage
-	if err := gc.imgStore.PutIndexContent(repo, index); err != nil {
+	// apply tags retention
+	if err := gc.retainTags(repo, &index); err != nil {
 		return err
 	}
 
+	if !gc.opts.DryRun {
```

Comment:
Do we have enough logging for the case in which DryRun is true? To see exactly what would be deleted?

Reply:
yes, we have logs in the functions that removes manifests. here we don't update the index.json of repo so that manifests won't get removed by gc.blobs() (unreferenced blobs)